### ISSUE TITLE
False positive error when instantiating `type[Self]` in abstract class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6184]
See this bug reported in the mypy issue tracker for details: https://github.com/python/mypy/issues/16274

Here's a minimal repro:

```python
from abc import ABC

class AbstrClassA(ABC):
    ...

v1 = [subclass() for subclass in AbstrClassA.__subclasses__()] # Should not generate an error
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
